### PR TITLE
Bulk Mailer

### PR DESCRIPTION
**Pull request by tabascq**
Add a bulk mailer for all players and for all team primary contacts. Currently the mailer is under strict admin control in order to make sure that we have a coordinated sender who knows about our monthly limits.

### CODE REVIEW

File: ServerCore/Pages/Events/Mailer.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                    foreach (string p in primaries)
+                    {
+                        primariesSplit.AddRange(p.Split(',', ';'));
+                    }
+
+                    addresses = primariesSplit;
+                    break;
+            }
+
+            MailHelper.Singleton.SendPlaintextBcc(addresses, MailSubject, MailBody);
```
Comment (morganbr) [inline ServerCore/Pages/Events/Mailer.cshtml.cs]: If addresses is empty, we probably shouldn't try to send

Comment (morganbr) [inline ServerCore/Pages/Events/Mailer.cshtml.cs]: Did we end up with an async version of sending mail? If so, it's probably better to use it for large batches

File: ServerCore/Pages/Events/Mailer.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            IEnumerable<string> addresses = Enumerable.Empty<string>();
+
+            switch(Group)
+            {
+                case MailGroup.AllPlayers:
```
Comment (morganbr) [inline ServerCore/Pages/Events/Mailer.cshtml.cs]: I suspect we also need a way to email small teams since we have a lot of them.

Comment (morganbr) [inline ServerCore/Pages/Events/Mailer.cshtml.cs]: For PuzzleDay we may need to email individual teams (e.g. "Go to the live event that you forgot"). I can't immediately think of a case for Hunt though.

File: ServerCore/Pages/Events/Mailer.cshtml.cs (hunk 1 above)
Comment (tabascq) [reply ServerCore/Pages/Events/Mailer.cshtml.cs]: Not going to worry about this edge case right now, it's not going to happen in practice for PH20

Comment (tabascq) [reply ServerCore/Pages/Events/Mailer.cshtml.cs]: MailHelper calls SendMailAsync() and does not await the result.

File: ServerCore/Pages/Events/Mailer.cshtml.cs (hunk 2 above)
Comment (tabascq) [reply ServerCore/Pages/Events/Mailer.cshtml.cs]: Added the ability to mail a team.

Comment (tabascq) [reply ServerCore/Pages/Events/Mailer.cshtml.cs]: Added the ability to mail a team.

**Pull request merged**